cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1872,6 +1872,105 @@ def get_distances(
             for _node in (self.leaves if leaves_only else self.nodes)
         }
 
+    def scale_to_unit_length(self) -> None:
+        """
+        Scales the tree to have unit length.
+
+        The longest path from root to leaf will have length 1 after the
+        scaling.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        current_times = self.get_times().values()
+        max_time = max(current_times)
+        min_time = min(current_times)
+        new_times = {}
+        for node in self.nodes:
+            new_times[node] = self.get_time(node) / (max_time - min_time)
+        self.set_times(new_times)
+
+    def get_depth(self) -> float:
```

Comment:
How is this function different from `get_max_depth_of_tree`?

Reply:
Ah this is really interesting:

1) I hadn't noticed that the method `get_max_depth_of_tree` already existed.
2) If the root has time different from 0, `get_max_depth_of_tree` and `get_depth` have different behavior. I believe this is a bug in `get_max_depth_of_tree`.

My proposal is that we remove `get_depth` and fix `get_max_depth_of_tree`.

Reply:
Yes I agree.

Just so we're on the same page -- the bug you're referring to is the fact that `get_max_depth_of_tree` need not return the maximum distance _from the root_ but just the maximum time of any leaf?

Then, the fix should be pretty easy -- we just want to replace the list comprehension statement with:

```
root_time = self.get_time(self.root)
depths = [self.get_time(l) - root_time for l in self.leaves]
```

Reply:
Yes, I agree!